.github/workflows/publish-new-version.yml

```diff
@@ -0,0 +1,106 @@
+name: Publish to S3 and PyPI
+
+on:
+  workflow_run:
+    workflows: ["Run Code Checks"]
+    branches: [develop]
+    types:
+      - completed
+
+jobs:
+  bump_version:
+    if: |
+        github.repository == 'opendatacube/odc-stats'
+        && github.event.workflow_run.conclusion == 'success'
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Push and tag versions
+        id: push-new-tag
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_ACTOR: autoaction
+        run: |
+          CTAG=$(git ls-remote --tags --sort='version:refname' | \
+            tail -1 | cut -d '/' -f3)
+          TAG=$(cat odc/stats/_version.py  | sed 's/[^0-9|.]//g')
+          if [[ $TAG == $CTAG || -z $CTAG ]]; then
+          git tag $TAG
+          git push origin --tags
+          echo "NEW=true" >> $GITHUB_OUTPUT
+          fi
+    outputs:
+      new_version: ${{ steps.push-new-tag.outputs.NEW }}
+
+  publish-pypi:
+    needs: [bump_version]
+    runs-on: ubuntu-latest
+    if: |
+      (needs.bump_version.result == 'success') &&
+      (needs.bump_version.outputs.new_version)
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v2
+        id: wheels_cache
+        with:
+          path: ./wheels
+          key: wheels-${{ github.sha }}
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: Install Twine
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools
+          python -m pip install --upgrade \
+           toml \
+           wheel \
+           twine
+          python -m pip freeze
+
+      - name: Upload to PyPI
+        env:
+          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          TWINE_USERNAME: __token__
+
+        run: |
+          ls wheels/clean/
+          twine upload --non-interactive --skip-existing wheels/clean/*
+
+  publish-s3:
+    needs: [bump_version]
+    runs-on: ubuntu-latest
+    if: |
+      (needs.bump_version.result == 'success') &&
+      (needs.bump_version.outputs.new_version)
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v2
+        id: wheels_cache
+        with:
+          path: ./wheels
+          key: wheels-${{ github.sha }}
+
+      - name: Prepare for upload to S3
+        run: |
+          mkdir -p ./pips
+          ./scripts/mk-pip-tree.sh ./wheels/clean/ ./pips
+          find ./pips -type f
+
+      - name: Upload to S3
+        run: |
+          aws s3 ls "${S3_DST}"
+          aws s3 sync ./pips/ "${S3_DST}"
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```

Comment:
Can this be done via role? (e.g., like ECR pushing)

Reply:
can, once the terraform pipeline fixed I can add more access to `github-actions-role`. Not saying it wasn't but I haven't checked